Fixes issue 2229: Push updated block headers to stoa

hewison-chris opened the pull request:
When missing signatures are added during catchup then the updated headers are sent to the configured addresses.

hewison-chris committed:
Simplify config for event handlers

This makes the code for the handlers more DRY. This is in preparation to
adding a new handler for the block header which will be pushed to stoa
when signatures are added.

Geod24 commented on source/agora/network/Manager.d:
```diff
@@ -1197,78 +1197,16 @@ public class NetworkManager
 
-    public TransactionReceivedHandler getTransactionReceivedHandler
-        (Address address)
+    public Handler getHandler (Handler)(Address address)
```
That's not possible. Templated function are not virtual.

hewison-chris replied:
Reverted back to handler for each but extracted repetition into new function `getRestInterfaceSettings`

Geod24 commented on source/agora/test/Base.d:
```diff
@@ -1267,22 +1267,7 @@ public class TestNetworkManager : NetworkManager
-    ///
-    protected final override TransactionReceivedHandler getTransactionReceivedHandler
-        (Address address)
+    protected final override auto getHandler (HandleType)(Address address)
```
The compiler should have screamed at you for using `override` here. Bad compiler.

hewison-chris replied:
Reverted

Geod24 commented on source/agora/common/Config.d:
```diff
@@ -1068,7 +1065,10 @@ noexist_event_handlers:
         auto node = Loader.fromString(conf_example).load();
         auto handlers = parserEventHandlers("event_handlers" in node, cmdln);
-        assert(handlers.length == 0);
+        assert(handlers[0].handler_addresses.length == 0);
+        assert(handlers[1].handler_addresses.length == 0);
+        assert(handlers[2].handler_addresses.length == 0);
+        assert(handlers[3].handler_addresses.length == 0);
```
`foreach` + keep the original `assert` ?

hewison-chris replied:
Updated

Geod24 reviewed (changes requested):
First commit doesn't work.

hewison-chris committed:
Push updated block headers to stoa

When missing signatures are added during catchup then the updated
headers are sent to the configured addresses.
